web/src/components/shared/bestComponent/BestComponent.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useContext, useState } from 'react';
+import { PageResourceContext } from '../../../contexts';
+import { SaleItem } from '../saleItem/SaleItem';
+import './style.css';
+
+const BestComponent = (): JSX.Element => {
+    const {
+        pageResource: { products },
+    } = useContext(PageResourceContext);
+    const [limit, setLimit] = useState(6);
+
+    return (
+        <div className="best__container">
+            <img
+                className="best__image"
+                src="https://d-art.ppstatic.pl/kadry/k/r/1/9d/0a/6234b5b877108_o_full.jpg"
+                alt="photo"
+            ></img>
+
+            <div className="best__items">
+                {products
+                    .slice(0, limit ? limit : products.length)
```

Comment:
limit is always, you dont have to check if it exists
AND
you should first filter to get products with the highest stars then slice array. The best way is to filter products that have more then 2 stars. Then sort them by stars number. Then slice the array. Then map it